### PR TITLE
[libc] final edits to newheadergen yaml files

### PR DESCRIPTION
**Pull request by RoseZhang03**
- final run of integration tests to deal with incorrect YAML input (finished sys headers, will finish the rest today)
- add any new functions made in recent PRs

### CODE REVIEW

File: libc/newhdrgen/yaml/sys/sys_time.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 header: sys-time.h
 standards: Linux
 macros: []
-types:
-  - type_name: struct_timeval
```
Comment (michaelrj-google) [inline libc/newhdrgen/yaml/sys/sys_time.yaml:5]: why was this removed?

File: libc/CMakeLists.txt [hunk 2]
```diff
@@ -72,6 +72,7 @@ option(LIBC_BUILD_GPU_LOADER "Always build the GPU loader utilities" OFF)
 if(LIBC_BUILD_GPU_LOADER OR (LLVM_LIBC_GPU_BUILD AND NOT LLVM_RUNTIMES_BUILD))
   add_subdirectory(utils/gpu)
 endif()
+option(LIBC_USE_NEW_HEADER_GEN "Generate header files using new headergen instead of the old one" ON)
```
Comment (michaelrj-google) [inline libc/CMakeLists.txt:75]: for now let's make the default `OFF` so that we can land this, then test it on the appropriate systems, then finally change the default.

File: libc/newhdrgen/yaml/sys/sys_time.yaml (hunk 1 above)
Comment (RoseZhang03) [reply libc/newhdrgen/yaml/sys/sys_time.yaml:5]: struct_timeval is already in the .h.def file, so becomes duplicated on the header file output

File: libc/CMakeLists.txt (hunk 2 above)
Comment (RoseZhang03) [reply libc/CMakeLists.txt]: Understood, I will change this on Aaryan's branch

**Review by michaelrj-google — APPROVED** [review]
LGTM
